### PR TITLE
Implement fast integer power routines in fluid properties

### LINKED ISSUE #11048: Implement MathUtils::pow in fluid properties where appropriate
## Rationale
There are a lot of places in the `fluid_properties` module where values are raised to integer powers, which I have been informed can be calculated a lot faster using `MathUtils::pow` rather than `std::pow`.

## Description
Replace `std::pow` with `MathUtils::pow` when raising values to integer powers in the `fluid_properties` module.

## Impact
Internal optimisation only

### PR DESCRIPTION
**Pull request by cpgr**
Swap std::pow with Utility:pow for fixed exponents, and with
MathUtils::pow when used in for loops.

Closes #11048

### CODE REVIEW

**Comment by cpgr** [comment]
Don't merge this yet - I'll rebase onto #11052 when it gets merged so that I can replace some of the remaining `std::pows` in these files where there is a negative integer exponent.

**Comment by cpgr** [comment]
OK, I've updated the powers with negative integer exponents as well now that #11052 is in, so it is ready to be looked at

File: modules/fluid_properties/include/userobjects/CO2FluidProperties.h [hunk 1]
```diff
@@ -309,49 +309,50 @@ class CO2FluidProperties : public SinglePhaseFluidPropertiesPT
...
-                        2.0, 2.0, 3.0, 3.0, 3.0, 4.0, 4.0, 4.0, 4.0, 4.0, 4.0, 5.0, 6.0};
-  std::vector<Real> _n3{
+  const std::vector<unsigned int> _d2{1, 2, 4, 5, 5, 5, 6, 6, 6, 1, 1,  4, 4, 4,
+                                      7, 8, 2, 3, 3, 5, 5, 6, 7, 8, 10, 4, 8};
+  const std::vector<Real> _t2{1.5,  1.5,  2.5,  0.0,  1.5,  2.0, 0.0, 1.0,  2.0,
+                              3.0,  6.0,  3.0,  6.0,  8.0,  6.0, 0.0, 7.0,  12.0,
+                              16.0, 22.0, 24.0, 16.0, 24.0, 8.0, 2.0, 28.0, 14.0};
+  const std::vector<unsigned int> _c2{1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2,
+                                      2, 2, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 6};
+  const std::vector<Real> _n3{
```
Comment (permcody) [inline modules/fluid_properties/include/userobjects/CO2FluidProperties.h]: Not to over-optimize, but you might start using `std::array` for these const fixed sized arrays. Maybe the compiler is doing something smart, but I'm not sure. `std::array` is a new C++11 type that's for the special case where you have a statically sized array. The compiler can allocate that space on the stack as opposed to the heap (which is where normal `std::vectors` are allocated). This subtle difference can make a big difference in speed. Take a look when you get a chance.

**Review by permcody — APPROVED** [review]

File: modules/fluid_properties/include/userobjects/CO2FluidProperties.h (hunk 1 above)
Comment (cpgr) [reply modules/fluid_properties/include/userobjects/CO2FluidProperties.h]: Thanks Cody, I'll change these over at some stage

**Pull request merged**